### ISSUE TITLE
Customized descriptions for any field + increase a description limit

### ISSUE THREAD

**Issue by david-kn** [nautobot/nautobot#1016]
### Environment
* Python version: any
* Nautobot version: 1.1.4

### Proposed Functionality

1. Ability to write a user defined (customized) descriptions for any kind of field (not just custom fields but also the regular ones e.g. role).  
![image](https://user-images.githubusercontent.com/14314517/138046549-fd8c1dad-2b56-4ab7-b7b0-c8ce10f2cd8f.png)

2. Extending the current description limit to more than 200 characters.

### Use Case
ad 1. The user can see right away a brief description and doesn't have to switch and search documentation or seek for help elsewhere when filling in fields (e.g. creating a prefix or modifying device). 
In an ideal world, the field itself together with options should be self-explanatory and clear but taking into consideration that with a complex network infrastructure behind and/or different teams and owners working with the system, not everyone can see or think of the consequences of each option value. 

ad 2. The 200 characters can be limiting when trying to write a structured (with css) or more informative description.

To sum it up and given the perspective I'm writing this issue with, it would improve user experience and could become a simple heads-up to understand what is meant and what can be an implication of setting some options to objects.

### Database Changes
( ? - add description columns for some already existing fields)

### External Dependencies
None

This is something in between a feature request and maybe housekeeping kind of thing? :-)

**Comment by jathanism** [comment]
> Hi Jathan, thanks for this information and nice overview. I see and understand your point of view and the complexity which it could/would bring.
> 
> Regarding the styling feature within `help_text` texts. Frankly, I was also personally surprised (especially with the js :-) ) that this is possible. But as the basic styling helps mainly users (I mean simple css is more than enough). Anyway - no matter the syntax (css, markdown, tex etc.) if this kind of feature would remain, it would be nice to hear... And, I guess, no one would really need nothing more than a simple styling capability (as bold, italic or bullet kind of thing which is doable in markdown with no problems).
> 
> To understand better you summary/suggestion:
> 
> > Introduce support for Markdown in CustomField.description to support custom rendering
> 
> If I got it correctly you mean CustomField `help_text` which would can be displayed below each field in form, right? Or/and also `description` which is shown in GUI in columns or when some record is open in detail view. I guess, having the ability to render markdown in both could be also interesting (so far I was thinking of styling the `help_text` only :-) ).

That is correct. This was implemented in the `next` (pre-1.2.0) branch in #1018 

> > Add help_text to every field on every object and every form field
> 
> You mean that all the fields from the core (e.g., platform, status) would have also some `help_text` visible below their fields (and would contain some default text describing the field). And the custom fields would remain as they are - definable `help_text` field in custom field setting?

That is correct! :)

**Comment by jathanism** [comment]
Per the suggested steps below we're gonna close this issue in exchange for these:
 
- Introduce support for Markdown in CustomField.description to support custom rendering (#1018)
- Update the label for description on Custom Fields to help_text to reduce confusion (#1049)
- Add help_text to every field on every object and every form field (#1050)
